robotiq_hande_driver/hardware/include/robotiq_hande_driver/protocol_logic.hpp

```diff
@@ -0,0 +1,371 @@
+#ifndef ROBOTIQ_HANDE_DRIVER__PROTOCOL_LOGIC_HPP_
+#define ROBOTIQ_HANDE_DRIVER__PROTOCOL_LOGIC_HPP_
+
+#include <string>
+
+#include "communication.hpp"
+
+
+namespace robotiq_hande_driver
+{
+
+/* Register mapping done based on Hand-E documentation:
+ * https://assets.robotiq.com/website-assets/support_documents/document/Hand-E_Instruction_Manual_e-Series_PDF_20190306.pdf
+ */
+enum class ActionRequestPositionBit : uint8_t {
+    ACTIVATE = 0u,                      /* rACT */
+    GO_TO = 3u,                         /* rGTO */
+    AUTOMATIC_RELEASE = 4u,             /* rATR */
+    AUTOMATIC_RELEASE_DIRECTION = 5u,   /* rARD */
+};
+
+enum class Activate : uint8_t {
+    DEACTIVATE_GRIPPER = 0u,
+    ACTIVATE_GRIPPER
+};
+
+enum class GoTo : uint8_t {
+    STOP = 0u,
+    GO_TO_REQ_POS
+};
+
+enum class AutomaticRelease : uint8_t {
+    NORMAL = 0u,
+    EMERGENCY_AUTO_RELEASE
+};
+
+enum class AutoReleaseDirection : uint8_t  {
+    CLOSING = 0u,
+    OPENING
+};
+
+/* Gripper Status */
+enum class ResponseByte : uint8_t {
+    STATUS = 0u,
+    FAULT_STATUS = 2u,
+    SPEED = 4u,
+    FORCE = 5u,
+};
+
+enum class StatusPositionBit : uint8_t {
+    ACTIVATION_STATUS = 0u,         /* gACT */
+    ACTION_STATUS = 3u,             /* gGTO */
+    GRIPPER_STATUS = 4u,            /* gSTA */
+    OBJECT_DETECTION_STATUS = 6u,   /* gObj */
+};
+
+constexpr auto ACTIVATION_STATUS_BITS = 0b1;
+enum class ActivationStatus : uint8_t {
+    GRIPPER_RESET = 0u,
+    GRIPPER_ACTIVATION
+};
+
+constexpr auto ACTION_STATUS_BITS = 0b1;
+enum class ActionStatus : uint8_t {
+    STOPPED = 0u,
+    GO_TO_POSITION_REQUEST
+};
+
+constexpr auto GRIPPER_STATUS_BITS = 0b11;
+enum class GripperStatus : uint8_t {
+    GRIPPER_IN_RESET = 0u,
+    ACTIVATION_IN_PROGRESS,
+    NOT_USED,
+    ACTIVATION_COMPLETE
+};
+
+constexpr auto OBJECT_DETECTION_STATUS_BITS = 0b11;
+enum class ObjectDetectionStatus : uint8_t {
+    MOTION_NO_OBJECT = 0u,
+    STOPPED_OPENING_DETECTED,
+    STOPPED_CLOSING_DETECTED,
+    REQ_POS_NO_OBJECT
+};
+
+constexpr auto GRIPPER_POSITION_OPENED_THRESHOLD = 230;
+constexpr auto GRIPPER_POSITION_CLOSED_THRESHOLD = 13;
+
+/**
+ * @brief This class contains protocol oriented functions and definitions.
+ */
+class ProtocolLogic{
+public:
+    ProtocolLogic();
+
+    ~ProtocolLogic() {};
+
+    /**
+     * @brief Initializes driver parameters.
+     *
+     * @param tty_port Modbus virtual port.
+     * @param baudrate Modbus serial baudrate.
+     * @param parity Modbus serial parity.
+     * @param data_bits Modbus serial data bits.
+     * @param stop_bit Modbus serial stopbit.
+     * @param slave_id Modbus slave id.
+     * @return None.
+     * @note The status should be checked to verify successful execution. An exception is thrown if communication issues occur.
+     */
+    void initialize(std::string tty_port, int baudrate, char parity, int data_bits, int stop_bit, int slave_id) {
```

Comment:
Let's reduce the double coping of the string
```suggestion
    void initialize(const std::string& tty_port, int baudrate, char parity, int data_bits, int stop_bit, int slave_id) {
```

Reply:
Done.